src/main/java/de/tum/in/www1/artemis/repository/ExamRepository.java

```diff
@@ -19,6 +21,13 @@
 
     List<Exam> findByCourseId(Long courseId);
 
+    @EntityGraph(type = LOAD, attributePaths = { "exerciseGroups" })
+    Optional<Exam> findWithExerciseGroupsById(Long id);
+
     @EntityGraph(type = LOAD, attributePaths = { "registeredUsers" })
     Optional<Exam> findWithRegisteredUsersById(Long id);
+
+    @EntityGraph(type = LOAD, attributePaths = { "exerciseGroups", "studentExams" })
+    @Query("select distinct exam from Exam exam left join fetch exam.studentExams studentExams left join fetch exam.exerciseGroups exerciseGroups left join fetch exerciseGroups.exercises where (exam.id =:#{#examId})")
```

Comment:
Using both @EntityGraph and @Query is redundant. We should be able to delete one.